federation/src/main/scala/caliban/federation/FederationDirectivesV1.scala

```diff
@@ -0,0 +1,15 @@
+package caliban.federation
+
+import caliban.Value.StringValue
+import caliban.parsing.adt.Directive
+import caliban.schema.Annotations.GQLDirective
+
+trait FederationDirectivesV1 {
+
+  case class GQLKey(fields: String) extends GQLDirective(Key(fields))
+
+  object Key {
+    def apply(fields: String, name: String = "key"): Directive =
+      Directive(name, Map("fields" -> StringValue(fields)))
+  }
+}
```

Comment:
v1 `@key` does not allow the `resolvable` argument, which could be a source of confusion to users. Splitting this so that using v1 no longer gives that option.